### PR TITLE
fix: bump secrets manager and IBM provider

### PR DESCRIPTION
**Pull request by shemau**
### Description

Bump IBM Provider to 1.62.0 and update secrets manager above 1.5.0 which required service_endpoints to be renamed allowed_network.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Bump IBM Provider to 1.62.0

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by alex-reiff — DISMISSED** [review]

File: version.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
     # Use "greater than or equal to" range in modules
     ibm = {
       source  = "IBM-Cloud/ibm"
-      version = ">= 1.51.0"
+      version = ">= 1.62.0"
```
Comment (ocofaigh) [inline version.tf]: Hmm, this should be `>= 1.62.0, <2.0.0` like our other modules. Must have been missed when that was rolled out.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
1 minor comment

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**